modules/analysis-common/build.gradle

```diff
@@ -25,11 +25,18 @@ dependencies {
   compileOnly project(':modules:lang-painless')
 }
 
+def v7compatibilityNotSupportedTests = {
+  return [
+     //marked as not needing compatible api
+     'indices.analyze/10_analyze/htmlStrip_deprecated', // Cleanup versioned deprecations in analysis #41560
+     'analysis-common/40_token_filters/delimited_payload_filter_error', //Remove preconfigured delimited_payload_filter #43686
+     'analysis-common/20_analyzers/standard_html_strip' // Cleanup versioned deprecations in analysis #41560
+  ]
+}
+
 tasks.named("yamlRestCompatTest").configure {
   systemProperty 'tests.rest.blacklist', [
-    'search.query/50_queries_with_synonyms/Test common terms query with stacked tokens',
-    'indices.analyze/10_analyze/htmlStrip_deprecated',
-    'analysis-common/40_token_filters/delimited_payload_filter_error',
-    'analysis-common/20_analyzers/standard_html_strip'
-  ].join(',')
+    'search.query/50_queries_with_synonyms/Test common terms query with stacked tokens', // #42654 - not sure, remove `common` query
```

Comment:
there is more tests that could be fixed by having https://github.com/elastic/elasticsearch/pull/42654 available under compatible api.
@jimczi this item is marked as `not_sure`. Do you have any views on implementing compatible api for it?